recipes/hpp-fcl/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+
+required_conan_version = ">=1.53.0"
+
+
+class HppFclConan(ConanFile):
+    name = "hpp-fcl"
+    description = "An extension of the Flexible Collision Library"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/humanoid-path-planner/hpp-fcl"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_py_interface": [True, False],
+        "install_documentation": [True, False],
+        "turn_assert_into_exception": [True, False],
+        "enable_logging": [True, False],
+        "has_qhull": [True, False],
+        "has_octomap": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_py_interface": False,
+        "install_documentation": False,
+        "turn_assert_into_exception": False,
+        "enable_logging": False,
+        "has_qhull": False,
+        "has_octomap": False
+    }
+
+    def requirements(self):
+        self.requires("eigen/[>=3.4.0]", package_id_mode="minor_mode")
+        self.requires("boost/1.76.0")
+        self.requires("assimp/5.4.1")
+        if self.options.has_octomap:
+            self.requires("octomap/1.10.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_PYTHON_INTERFACE"] = self.options.with_py_interface
+        tc.variables["INSTALL_DOCUMENTATION"] = self.options.install_documentation
+        tc.variables["HPP_FCL_ENABLE_LOGGING"] = self.options.enable_logging
+        tc.variables["HPP_FCL_HAS_OCTOMAP"] = self.options.has_octomap
+        if Version(self.version) < "2.4.5":
+            tc.variables["HPP_FCL_HAS_QHULL"] = self.options.has_qhull
+            del self.options.turn_assert_into_exception
+        else:
+            tc.variables["HPP_FCL_TURN_ASSERT_INTO_EXCEPTION"] = self.options.turn_assert_into_exception
+            del self.options.has_qhull
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
I decided to remove this line since it was shamelessly copied from other examples 😑

Reply:
not removing it is not enough for it to be used I guess. I think you should either:

1. keep that rmdir, but then you need to teach conan how to generate properly `hpp-fcl-config.cmake` with all required dependencies ; 
2. **or** keep that directory, but then you need to teach conan not to generate any `hpp-fcl-config.cmake`, and use the upstream-generated `lib/cmake/hpp-fclConfig.cmake` file instead, as it already contains the right dependencies.



For now, I can't build your package:

```c++
/home/gsaurel/.conan2/p/b/hpp-fba0c52d2f5443/p/include/hpp/fcl/data_types.h:41:10: fatal error: 'Eigen/Core' file not found
   41 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
```

This is because a not-good-enough `hpp-fcl-config.cmake` file was generated by conan. You can see this by adding :

```diff
--- a/recipes/hpp-fcl/all/test_package/CMakeLists.txt
+++ b/recipes/hpp-fcl/all/test_package/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.10)
 project(test_package LANGUAGES CXX)
 
 find_package(hpp-fcl REQUIRED CONFIG)
+message(STATUS "hpp-cfcl config file was found on ${hpp-fcl_DIR}")
 
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} PRIVATE hpp-fcl::hpp-fcl)
```

Which display: `-- hpp-cfcl config file was found on /tmp/conan-center-index/recipes/hpp-fcl/all/test_package/build/gcc-13-x86_64-gnu17-release/generators`

Reply:
😞 
I'm sorry...

I have to give up on this since even after reading up on the docs and trying things, I wasn't able to pull this out 😕 
https://docs.conan.io/2/examples/tools/cmake/cmake_toolchain/use_package_config_cmake.html